bin/convert2vulyk.py

```diff
@@ -42,116 +83,173 @@ def convert_bsf_2_vulyk(text: str, bsf_markup: str) -> dict:
-                    'or you can pipe data in and out of this script like:'
-                    '`cat file.txt | python3 convert2vulyk.py > save_to_file.json`')
-    parser.add_argument('text_file', nargs="?", type=argparse.FileType('r'), default=sys.stdin,
-                        help='Text file to process. If file path not provided, assumes stdin stream.'
-                             ' Must be tokenized if --brat_file is provided. '
-                             '*.ann for lang-uk data set')
-    parser.add_argument('--brat_annotation', type=argparse.FileType('r'), default=None,
-                        help='Path to file with Brat standoff markup')
-    parser.add_argument('-v', action='store_true', help='Print more logs (info)')
-    parser.add_argument('-vv', action='store_true', help='Print even more logs (debug)')
+        description="Convert tokenized pre-annotated text or tag generic text to json file supported by Vulyk. "
+        "Data file can be supplied via cmd line arguments "
+        "or you can pipe data in and out of this script like:"
+        "`cat file.txt | python3 convert2vulyk.py tag >> save_to_file.json`"
+    )
+
+    parser.add_argument(
+        "text_files",
+        help="File mask to collect text files to process."
+        " Must be tokenized if convert command is used. "
+        "*.ann for lang-uk data set",
+    )
+
+    parser.add_argument(
+        "-d",
+        "--debug",
+        help="Print even more logs (debug)",
+        action="store_const",
+        dest="loglevel",
+        const=logging.DEBUG,
+        default=logging.WARNING,
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        help="Print more logs (info)",
+        action="store_const",
+        dest="loglevel",
+        const=logging.INFO,
+    )
+
+    subparsers = parser.add_subparsers(help="Available commands")
+
+    convert_parser = subparsers.add_parser(
+        "convert",
+        help="Convert tokenized texts (with optional annotations in a separate files) to a Vulyk format",
+    )
+
+    convert_parser.set_defaults(func=convert)
+    convert_parser.add_argument(
+        "--ann_autodiscovery",
+        default="replace",
+        choices=("append", "replace"),
+        help="How to find *.ann files: by appending or replacing the extension",
+    )
+    convert_parser.add_argument(
+        "--ignore_annotations", default=False, action="store_true", help="Do not try to load *.ann files"
+    )
+
+    tag_parser = subparsers.add_parser("tag", help="Tag given files using one of frameworks/models specified in params")
+
+    tag_parser.add_argument(
+        "--ner_framework",
+        choices=("stanza", "spacy"),
+        default="stanza",
+        help="Which framework to use for the tagging",
+    )
+
+    tag_parser.add_argument(
+        "--ner_model",
+        default="uk",
+        help="Which model to use. Stanza has pre-built model `uk` for ukrainian language. "
```

Comment:
Stanza also requires language data to be downloaded before the first use. https://stanfordnlp.github.io/stanza/download_models.html

Reply:
The difference here is that Stanza has built-in command that you use (and I moved it to the different function).

For SpaCy the situation is different, you can download the model from their repo using the cli command, but it's cumbersome to call it from the code (and I clearly don't want to import their cli-related scripts).

If you know how to make stanza load custom model from the disk — please amend the code so it'll know when to download and when to load it from disk

Reply:
For stanza, you basically can provide everything in their init statement 

nlp = stanza.Pipeline('uk', processors='tokenize,pos,lemma', 
                      ner_model_path='your_path/saved_models/ner/uk_languk_nertagger.pt', 
                      ner_forward_charlm_path="", ner_backward_charlm_path="")

But I'm not sure how to replace download('language'). I'll check later on this.

Reply:
well, you might check if the `Path(args.ner_model).exists()` and if it's not you do the download and pass the different args to the constructor.